pinecone/client.go

```diff
@@ -273,6 +273,104 @@ func (c *Client) DeleteIndex(ctx context.Context, idxName string) error {
+//  or you could pass "p1.x4" to scale your index to the "x4" size, and
+//  so forth.
+//  - replicas: (Optional) The number of replicas to scale the index to.
+//   This is capped by the maximum number of replicas allowed in your Pinecone project. To configure this number,
+//   go to [app.pinecone.io], select your project, and configure the maximum number of pods.
+//
+// Note: You can only scale an index up, not down. If you want to scale an index down,
+// you must create a new index with the desired configuration.
+//
+// Returns a ConfigureIndexResponse object, which contains the new configuration of the index, or an error.
+//
+// Example for a pods-based index originally configured with 1 "p1" pod of size "x2" and 1 replica:
+//  // To scale the size of your pods from "x2" to "x4":
+//   _, err := pc.ConfigureIndex(ctx, "my-index", "p1.x4", nil)
+//   if err != nil {
+//       fmt.Printf("Failed to configure index: %v\n", err)
+//   }
+//
+//  // To scale the number of replicas:
+//   _, err := pc.ConfigureIndex(ctx, "my-index", nil, 4)
+//   if err != nil {
+//       fmt.Printf("Failed to configure index: %v\n", err)
+//   }
+//
+//  // To scale both the size of your pods and the number of replicas:
+//   _, err := pc.ConfigureIndex(ctx, "my-index", "p1.x4", 4)
+//   if err != nil {
+//       fmt.Printf("Failed to configure index: %v\n", err)
+//   }
+//
+// [scale a pods-based index]: https://docs.pinecone.io/guides/indexes/configure-pod-based-indexes
+// [app.pinecone.io]: https://app.pinecone.io
+func (c *Client) ConfigureIndex(ctx context.Context, name string, pods *string,
+	replicas *int32) (*control.ConfigureIndexResponse,
+	error) {
+
+	var podType *control.PodSpecPodType
+	var replicasAmt *control.PodSpecReplicas
+
+	if pods == nil && replicas == nil {
+		return nil, fmt.Errorf("Must specify either pods or replicas")
+	}
+
+	if pods != nil {
+		podTypeVal := control.PodSpecPodType(*pods)
+		podType = &podTypeVal
+	}
+	if replicas != nil {
+		replicasVal := control.PodSpecReplicas(*replicas)
+		replicasAmt = &replicasVal
+	}
+
+	request := control.ConfigureIndexRequest{
+		Spec: struct {
+			Pod struct {
+				PodType  *control.PodSpecPodType  `json:"pod_type,omitempty"`
+				Replicas *control.PodSpecReplicas `json:"replicas,omitempty"`
+			} `json:"pod"`
+		}{
+			Pod: struct {
+				PodType  *control.PodSpecPodType  `json:"pod_type,omitempty"`
+				Replicas *control.PodSpecReplicas `json:"replicas,omitempty"`
+			}{
+				PodType:  podType,
+				Replicas: replicasAmt,
+			},
+		},
+	}
+
+	req, err := c.restClient.ConfigureIndex(ctx, name, request)
+	if err != nil {
+		log.Fatalf("Failed to configure index %s. Error: %v", name, err)
+	}
+	if req.StatusCode != http.StatusOK {
+		err := handleErrorResponseBody(req, "failed to configure index: ")
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	response, err := control.ParseConfigureIndexResponse(req) // TODO: need this?
```

Comment:
I don't think we should be using the internals, I don't think we're doing that anywhere else. I'd maybe look at how we're handling things in something like `DescribeIndex`. Could you juse use the `decodeIndex` function instead?

```suggestion
	res, err := c.restClient.ConfigureIndex(ctx, name, request)
		if err != nil {
		log.Fatalf("Failed to configure index %s. Error: %v", name, err)
	}
	defer res.Body.Close()
	
	if req.StatusCode != http.StatusOK {
	    return nil, handleErrorResponseBody(req, "failed to configure index: ")
	}
	
	return decodeIndex(res.Body)
```

Ultimately this just returns an `IndexModel`, there's not really anything else in `ConfigureIndexResponse`, so you'd need to update the return type and maybe doc string here.

Reply:
Sure yeah good call, I'll do that!